### PR TITLE
feat(collections/sort_by): descending order can be specified in options

### LINKED ISSUE #3397: Should sort_by support descending order?
### Discussed in https://github.com/denoland/deno_std/discussions/3393

<div type='discussions-op-text'>

<sup>Originally posted by **sant123** May 22, 2023</sup>
Hi, I see that `collections/sortBy` function orders by ascending. Could it be possible to support descending? 

Also not sure if another overloading would be good or creating a new function for this.

Thanks.</div>

I can create a PR with a new function called `collections/sortByDesc`. Please let me know 🙏🏼

### PR DESCRIPTION
**Pull request by sant123**
Closes #3397

### CODE REVIEW

**Comment by kt3k** [comment]
I think it's too verbose to duplicate all test cases for desc ordering (Also it's waste of CI time). I suggest we should add a single test case with title `[collections/sortBy] desc ordering` which exercises one or two typical cases.

File: collections/sort_by.ts [hunk 1]
```diff
@@ -64,7 +97,7 @@ export function sortBy<T>(
     const b = selectors[bi];
     if (a === null) return 1;
     if (b === null) return -1;
```
Comment (kt3k) [inline collections/sort_by.ts:99]: Do these lines also need to be changed?

How about swapping `a` and `b` above if the order is "desc"? (In that case I think we don't need to switch `comparator`)

Comment (sant123) [reply collections/sort_by.ts]: Now they are swapped, good catch! Also the idea of `comparator` is to extract the function ordering once and call it when `a` and `b` are not null. However I do not understand why coverage says there's no coverage in `comparator` 🤔

File: collections/sort_by.ts [hunk 2]
```diff
@@ -48,23 +68,39 @@ export function sortBy<T>(
...
+
   indexes.sort((ai, bi) => {
     const a = selectors[ai];
     const b = selectors[bi];
-    if (a === null) return 1;
-    if (b === null) return -1;
-    return a > b ? 1 : a < b ? -1 : 0;
+    if (a === null) return aIsNull;
+    if (b === null) return bIsNull;
+    return comparator(a, b);
```
Comment (kt3k) [inline collections/sort_by.ts:103]: How about swapping a, b like the below? The current implementation looks hard to follow to me..

```js
  indexes.sort((ai, bi) => {
    let a = selectors[ai];
    let b = selectors[bi];
    if (order === "desc") {
      [a, b] = [b, a];
    }
    if (a === null) return 1;
    if (b === null) return -1;
    return a > b ? 1 : a < b ? -1 : 0;
  });
```

Comment (sant123) [reply collections/sort_by.ts]: Definitely better, now got what you meant. Thank you so much @kt3k!

**Review by kt3k — APPROVED** [review]
LGTM
